src/Microsoft/Provider.php

```diff
@@ -58,20 +64,39 @@ protected function getTokenUrl()
      */
     protected function getUserByToken($token)
     {
-        $response = $this->getHttpClient()->get(
+        $responseUser = $this->getHttpClient()->get(
             'https://graph.microsoft.com/v1.0/me',
             [
                 RequestOptions::HEADERS => [
                     'Accept'        => 'application/json',
                     'Authorization' => 'Bearer '.$token,
                 ],
                 RequestOptions::QUERY => [
-                    '$select' => implode(',', array_merge(self::DEFAULT_FIELDS, ($this->config['fields'] ?: []))),
+                    '$select' => implode(',', array_merge(self::DEFAULT_FIELDS_USER, ($this->config['fields'] ?: []))),
                 ],
             ]
         );
 
-        return json_decode((string) $response->getBody(), true);
+        $formatted_response = json_decode((string) $responseUser->getBody(), true);
+
+        if (($this->getConfig('tenant', 'common') === 'common') && ($this->getConfig('include_tenant_info', false) === true)) {
+            $responseTenant = $this->getHttpClient()->get(
+                'https://graph.microsoft.com/v1.0/organization',
+                [
+                    RequestOptions::HEADERS => [
+                        'Accept'        => 'application/json',
+                        'Authorization' => 'Bearer '.$token,
+                    ],
+                    RequestOptions::QUERY => [
+                        '$select' => implode(',', array_merge(self::DEFAULT_FIELDS_TENANT, ($this->config['tenant_fields'] ?: []))),
+                    ],
+                ]
+            );
+
+            $formatted_response['tenant'] = json_decode((string) $responseTenant->getBody(), true)['value'][0];
```

Comment:
as above. Do we need a null coalesce here to make sure this does't error if the request doesn't have results?
```suggestion
            $formatted_response['tenant'] = json_decode((string) $responseTenant->getBody(), true)['value'][0] 
            ?? null;
```